module/VuFind/src/VuFind/ILS/Driver/CombinedDriver.php

```diff
@@ -0,0 +1,1346 @@
+    {
+        return $this->combineMultipleArraysOfAssociativeArrays('getStatuses', [$ids], 'id');
+    }
+
+    /**
+     * Get suppressed authority records
+     *
+     * @return array ID numbers of suppressed authority records in the system.
+     */
+    public function getSuppressedAuthorityRecords()
+    {
+        return $this->defaultCall('getSuppressedAuthorityRecords', func_get_args());
+    }
+
+    /**
+     * Get suppressed records.
+     *
+     * @throws ILSException
+     * @return array ID numbers of suppressed records in the system.
+     */
+    public function getSuppressedRecords()
+    {
+        return $this->defaultCall('getSuppressedRecords', func_get_args());
+    }
+
+    /**
+     * Provide an array of URL data (in the same format returned by the record
+     * driver's getURLs method) for the specified bibliographic record.
+     *
+     * @param string $id Bibliographic record ID
+     *
+     * @return array
+     */
+    public function getUrlsForRecord(string $id): array
+    {
+        return $this->defaultCall('getUrlsForRecord', func_get_args());
+    }
+
+    /**
+     * Has Holdings
+     *
+     * This is responsible for determining if holdings exist for a particular
+     * bibliographic id
+     *
+     * @param string $id The record id to retrieve the holdings for
+     *
+     * @return bool True if holdings exist, False if they do not
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function hasHoldings($id)
+    {
+        return $this->defaultCall('hasHoldings', func_get_args());
+    }
+
+    /**
+     * Get Hidden Login Mode
+     *
+     * This is responsible for indicating whether login should be hidden.
+     *
+     * @return bool true if the login should be hidden, false if not
+     */
+    public function loginIsHidden()
+    {
+        return $this->defaultCall('loginIsHidden', func_get_args());
+    }
+
+    /**
+     * Patron Login
+     *
+     * This is responsible for authenticating a patron against the catalog.
+     *
+     * @param string $username The patron barcode
+     * @param string $password The patron password
+     *
+     * @throws ILSException
+     * @return mixed           Associative array of patron info on successful login,
+     * null on unsuccessful login.
+     */
+    public function patronLogin($username, $password)
+    {
+        return $this->defaultCall('patronLogin', func_get_args());
+    }
+
+    /**
+     * Place Hold
+     *
+     * Attempts to place a hold or recall on a particular item and returns
+     * an array with result details
+     *
+     * @param array $holdDetails An array of item and patron data
+     *
+     * @return mixed An array of data on the request including
+     * whether or not it was successful and a system message (if available)
+     */
+    public function placeHold($holdDetails)
+    {
+        return $this->defaultCall('placeHold', func_get_args());
```

Comment:
This, like other methods that mention "a system message" in @return, should return an array such as `['success' => false, 'status' => 'ils_action_unavailable']` on failure.

Reply:
Now, an exception is thrown if the method is not supported.